### DISCUSSION TITLE
Do system messages support threading and reactions?

### DISCUSSION THREAD

**Discussion by jagged91** [GetStream/stream-chat-flutter#654] [Q&A] [answered]
This is not a bug, but a general question which isn't covered by the documentation. Our project involves some inventive features built on top of chat, which require system messages as certain events are triggered on our backend. To support this, we plan on using systemMessageBuilder to build completely custom UI into chat. 

If we implement our own threading and replies UI on some of these system messages, are we able to create our own threads and add replies?

Oh, and another question: Do system messages trigger 'unreads'?

**Comment by imtoori** [comment, marked as answer]
yes, they do, but you need to build your own UI at the moment, but I saw you're doing that 👍🏼 
Regarding unreads, they don't increase unread counts
